samples/openthread/cli/common.cmake

```diff
@@ -0,0 +1,97 @@
+
+include(${NRFXLIB_DIR}/common.cmake)
+
+# Copy new OpenThread libraries, headers, text file with the configuration
+# to the nrfxlib openthread folder.
+function(ot_libs_substitute)
+  execute_process(
+    COMMAND           git rev-parse HEAD
+    WORKING_DIRECTORY ${openthread_SOURCE_DIR}
+    RESULT_VARIABLE   git_return
+    OUTPUT_VARIABLE   openthread_git_hash
+    )
+
+  list(APPEND OPENTHREAD_SETTINGS "OpenThread_commit=${openthread_git_hash}\n\n")
+
+  # Store all OT related variables
+  get_cmake_property(_variableNames VARIABLES)
+  list (SORT _variableNames)
+  foreach (_variableName ${_variableNames})
+    if("${_variableName}" MATCHES "^CONFIG_OPENTHREAD_.*|^OT_.*")
+      list(APPEND OPENTHREAD_SETTINGS "${_variableName}=${${_variableName}}\n")
+    endif()
+  endforeach()
+
+  set(OPENTHREAD_CONFIG_FILE
+    "${PROJECT_BINARY_DIR}/openthread_lib_configuration.txt")
+  FILE(WRITE ${OPENTHREAD_CONFIG_FILE} ${OPENTHREAD_SETTINGS})
+
+  set(OPENTHREAD_SRC_DIR "${PROJECT_BINARY_DIR}/modules/openthread/build/src")
+  set(OPENTHREAD_LIBS
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/core/libopenthread-radio.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/cli/libopenthread-cli-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-ftd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-ncp-mtd.a"
+    "${OPENTHREAD_SRC_DIR}/ncp/libopenthread-rcp.a"
+  )
+
+  nrfxlib_calculate_lib_path(lib_path)
+
+  foreach(lib IN LISTS OPENTHREAD_LIBS)
+    add_custom_command(
+      TARGET app
+      COMMENT "Copying OpenThread library: ${lib}."
+      PRE_LINK
+      COMMAND ${CMAKE_COMMAND} -E copy "${lib}"
+        "${NRFXLIB_DIR}/openthread/${lib_path}/v${OT_THREAD_VERSION}"
+      COMMAND echo
+        "Updated nrfxlib OpenThread library with: ${lib}"
+      DEPENDS ${OPENTHREAD_LIBS}
+      )
+  endforeach()
+
+  add_custom_command(
+    TARGET app
+    COMMENT "Copying OpenThread headers."
+    PRE_LINK
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${NRFXLIB_DIR}/openthread/include/openthread-config-generic.h
+      ${PROJECT_BINARY_DIR}
+    COMMAND ${CMAKE_COMMAND} -E remove
+      ${openthread_SOURCE_DIR}/include/openthread-config-android.h
+    COMMAND ${CMAKE_COMMAND} -E remove
+      ${openthread_SOURCE_DIR}/include/Makefile.am
+    COMMAND ${CMAKE_COMMAND} -E remove
+      ${openthread_SOURCE_DIR}/include/openthread/Makefile.am
+    COMMAND ${CMAKE_COMMAND} -E remove
+      ${openthread_SOURCE_DIR}/include/openthread/platform/Makefile.am
+    COMMAND ${CMAKE_COMMAND} -E remove_directory -f
+      ${NRFXLIB_DIR}/openthread/include
+    COMMAND ${CMAKE_COMMAND} -E copy_directory
+      ${openthread_SOURCE_DIR}/include
+      ${NRFXLIB_DIR}/openthread/include
+    COMMAND ${CMAKE_COMMAND} -E copy
+      ${PROJECT_BINARY_DIR}/openthread-config-generic.h
+      ${NRFXLIB_DIR}/openthread/include
```

Comment:
I believe you just copied this file in the opposite direction some lines above.

Reply:
And I deleted the whole directory in the meantime.

Reply:
so if user had uncommited changes, e.g. new files, then it's just bad luck ?
The build system should not decide to completely remove a folder that could have local changes/new files.

Reply:
Good point. Implementing a separate target should fix that.